### PR TITLE
Implement NTLM and add test cases

### PR DESCRIPTION
**Pull request by harryi3t**

### CODE REVIEW

File: lib/authorizer/ntlm.js [hunk 1]
```diff
@@ -54,14 +55,13 @@ module.exports = {
      */
     post: function (auth, response, done) {
         var state = auth.get('state'),
-            domain = auth.get('domain'),
-            workstation = auth.get('workstation'),
+            domain = auth.get('domain') || '',
```
Comment (kunagpal) [inline lib/authorizer/ntlm.js]: Use a constant reference.

File: lib/authorizer/ntlm.js [hunk 2]
```diff
@@ -54,14 +55,13 @@ module.exports = {
...
+            workstation = auth.get('workstation') || '',
             username = auth.get('username'),
             password = auth.get('password'),
             typeOneMessage,
             typeTwoMessage,
-            typeThreeMessage,
-            item = context.item;
+            typeThreeMessage;
```
Comment (kunagpal) [inline lib/authorizer/ntlm.js]: Better variable naming?

File: test/integration/auth-methods/ntlm.test.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+describe('NTLM', function () {
+    var ntlmServer,
+        ntlmServerPort = 3456,
+        testrun,
+        runOptions = {
+            collection: {
+                item: {
+                    name: 'NTLM Sample Request',
+                    request: {
+                        url: 'http://localhost:' + ntlmServerPort,
```
Comment (kunagpal) [inline test/integration/auth-methods/ntlm.test.js]: Where is this server being started?

File: lib/authorizer/ntlm.js (hunk 2 above)
Comment (harryi3t) [reply lib/authorizer/ntlm.js]: type3Message?

File: test/integration/auth-methods/ntlm.test.js (hunk 3 above)
Comment (harryi3t) [reply test/integration/auth-methods/ntlm.test.js]: in the `before` block [line 79](https://github.com/postmanlabs/postman-runtime/pull/390/files#diff-941f706d7c82f0f8633eb82a743ba94bR79)
and stopped in `after` block

File: lib/authorizer/ntlm.js [hunk 4]
```diff
@@ -136,6 +142,14 @@ module.exports = {
      * @param {AuthHandlerInterface~authSignHookCallback} done
      */
     sign: function (auth, request, done) {
+        request.removeHeader(AUTHORIZATION, {ignoreCase: true});
+        if (auth.get(NTLM_HEADER)) {
+            request.addHeader({
+                key: AUTHORIZATION,
+                value: auth.get(NTLM_HEADER),
```
Comment (kamalaknn) [inline lib/authorizer/ntlm.js]: Redundant `get`

**Pull request merged**